WHG/Brain.js

```diff
@@ -44,7 +44,7 @@ class Brain {
   //-------------------------------------------------------------------------------------------------------------------------------------
   //returns a perfect copy of this brain object
   clone() {
-    var clone = new Brain(this.directions.length);
+    var clone = new Brain(0); // create zero-sized brain as it will be rewritten

```

Comment:
What do you mean by "it will be rewritten", if I may ask?

Reply:
Brain constructor initializes random directions, but we copy them from current brain immediately after that, so that initialization is a waste of time

Reply:
Oh, I didn't notice that. 
My bad.

Wouldn't it be better to create a default constructor, in that case?